system/lib/update_llvm_libc.py

```diff
@@ -0,0 +1,119 @@
+#!/usr/bin/env python3
+# Copyright 2019 The Emscripten Authors.  All rights reserved.
+# Emscripten is available under two separate licenses, the MIT license and the
+# University of Illinois/NCSA Open Source License.  Both these licenses can be
+# found in the LICENSE file.
+
+import os
+import sys
+import shutil
+import glob
+import subprocess
+
+script_dir = os.path.abspath(os.path.dirname(__file__))
+emscripten_root = os.path.dirname(os.path.dirname(script_dir))
+default_llvm_dir = os.path.join(os.path.dirname(emscripten_root), 'llvm-project')
+emscripten_patches = os.path.join(script_dir, "llvm-libc", "patches", "*.patch")
+
+preserve_files = ('readme.txt', '__assertion_handler', '__config_site')
+# ryu_long_double_constants.h from libc is unused (and very large)
+excludes = ('CMakeLists.txt', 'ryu_long_double_constants.h')
+
+libc_copy_dirs = [
+    'hdr',
+    'include/llvm-libc-macros',
+    'include/llvm-libc-types',
+    'shared',
+    'config',
+    'src/__support',
+    'src/string',
+    'src/strings',
+    'src/errno',
+    'src/math',
+    'src/stdlib',
+    'src/inttypes',
+    'src/stdio/printf_core',
+]
+
+libc_exclusion_patterns = [
+    'src/math/generic/*f16*', # float16 is unsupported in Emscripten.
+    'src/strings/str*casecmp_l*' # locale_t is unsupported in Overlay Mode.
+]
+
+def clean_dir(dirname):
+  if not os.path.exists(dirname):
+    return
+  for f in os.listdir(dirname):
+    if f in preserve_files:
+      continue
+    full = os.path.join(dirname, f)
+    if os.path.isdir(full):
+      shutil.rmtree(full)
+    else:
+      os.remove(full)
+
+
+def copy_tree(upstream_dir, local_dir):
+  if not os.path.exists(local_dir):
+    os.makedirs(local_dir)
+  for f in os.listdir(upstream_dir):
+    full = os.path.join(upstream_dir, f)
+    if os.path.isdir(full):
+      shutil.copytree(full, os.path.join(local_dir, f))
+    elif f not in excludes:
+      shutil.copy2(full, os.path.join(local_dir, f))
+  for root, dirs, files in os.walk(local_dir):
+    for f in files:
+      if f in excludes:
+        full = os.path.join(root, f)
+        os.remove(full)
+
+def apply_patch(patch_file):
+  try:
+    command = ["git", "apply", "-p2", "--directory=system/lib/llvm-libc/", patch_file]
+
+    result = subprocess.run(
+        command,
+        capture_output=True,
+        text=True,
+        check=True
+    )
+
+  except subprocess.CalledProcessError as e:
+    print(f"Error applying patch: {e}")
+    print("STDOUT:", e.stdout)
+    print("STDERR:", e.stderr)
```

Comment:
If you don't capture the output is the error message not clear enough on its own?   I.e. maybe jsut remove the whole try catch and the `capture_output` arg?